### PR TITLE
[9.x] Add dot to collection classes

### PR DESCRIPTION
**Pull request by hamidrezaniazi**
Add `dot` method to the collection classes. It's only available in the `Arr` class now.

### CODE REVIEW

File: src/Illuminate/Collections/Collection.php [hunk 1]
```diff
@@ -1515,6 +1515,16 @@ public function transform(callable $callback)
         return $this;
     }
 
+    /**
+     * Convert an expanded array into a flatten "dot" notation array.
```
Comment (GrahamCampbell) [inline src/Illuminate/Collections/Collection.php:1519]: typo. flattened.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Support/SupportCollectionTest.php [hunk 2]
```diff
@@ -4963,6 +4963,42 @@ public function testCollect($collection)
...
 
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testDot($collection)
+    {
+        $data = $collection::make([
+            'name' => 'Taylor',
+            'meta' => [
+                'foo' => 'bar',
```
Comment (GrahamCampbell) [inline tests/Support/SupportCollectionTest.php:4974]: Missing test coverage when the values themselves are also collections.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Collections/Collection.php (hunk 1 above)
Comment (hamidrezaniazi) [reply src/Illuminate/Collections/Collection.php]: 426af3b73

File: tests/Support/SupportCollectionTest.php (hunk 2 above)
Comment (hamidrezaniazi) [reply tests/Support/SupportCollectionTest.php:4974]: d46028748

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**